tests/integration/pipeline/test_bootstrap_command.py

```diff
@@ -0,0 +1,163 @@
+from unittest import skipIf
+
+from parameterized import parameterized
+
+from samcli.lib.pipeline.bootstrap.stage import Stage
+from tests.integration.pipeline.base import BootstrapIntegBase
+from tests.testing_utils import run_command_with_input, RUNNING_ON_CI, RUNNING_TEST_FOR_MASTER_ON_CI, RUN_BY_CANARY
+
+# bootstrap tests require credentials and CI/CD will only add credentials to the env if the PR is from the same repo.
+# This is to restrict tests to run outside of CI/CD, when the branch is not master or tests are not run by Canary
+SKIP_DEPLOY_TESTS = RUNNING_ON_CI and RUNNING_TEST_FOR_MASTER_ON_CI and not RUN_BY_CANARY
+
+
+@skipIf(SKIP_DEPLOY_TESTS, "Skip bootstrap tests in CI/CD only")
+class TestBootstrap(BootstrapIntegBase):
+    @parameterized.expand([("create_ecr_repo",), (False,)])
+    def test_interactive_with_no_resources_provided(self, create_ecr_repo: bool):
+        stage_name = self._method_to_stage_name(self.id())
+        self.stack_names = [Stage._get_stack_name(stage_name)]
+
+        bootstrap_command_list = self.get_bootstrap_command_list(interactive=True)
+
+        inputs = [
+            stage_name,
+            "",  # pipeline user
+            "",  # Pipeline execution role
+            "",  # CloudFormation execution role
+            "",  # Artifacts bucket
+            "2" if create_ecr_repo else "1",  # Should we create ECR repo, 1 - No, 2 - Yes
+            "",  # Pipeline IP address range
+            "y",  # proceed
+        ]
+
+        bootstrap_process_execute = self.run_command_with_inputs(bootstrap_command_list, inputs)
+
+        self.assertEqual(bootstrap_process_execute.process.returncode, 0)
+        stdout = bootstrap_process_execute.stdout.decode()
+        self.assertIn("We have created the following resources", stdout)
+        # make sure pipeline user's credential is printed
+        self.assertIn("ACCESS_KEY_ID", stdout)
+        self.assertIn("SECRET_ACCESS_KEY", stdout)
+        if create_ecr_repo:
+            self.assertIn("arn:aws:ecr:", stdout)
+            self.assertEqual(5, self._count_created_resources(stdout))
+        else:
+            self.assertNotIn("arn:aws:ecr:", stdout)
+            self.assertEqual(4, self._count_created_resources(stdout))
+
+    def test_interactive_with_all_required_resources_provided(self):
+        stage_name = self._method_to_stage_name(self.id())
+        self.stack_names = [Stage._get_stack_name(stage_name)]
+
+        bootstrap_command_list = self.get_bootstrap_command_list(interactive=True)
+
+        inputs = [
+            stage_name,
+            "arn:aws:iam::123:user/user-name",  # pipeline user
+            "arn:aws:iam::123:role/role-name",  # Pipeline execution role
+            "arn:aws:iam::123:role/role-name",  # CloudFormation execution role
+            "arn:aws:s3:::bucket-name",  # Artifacts bucket
+            "3",  # Should we create ECR repo, 3 - specify one
+            "arn:aws:ecr:::repository/repo-name",  # ecr repo
```

Comment:
[nit] it  worth adding an AWS region, i.e. "arn:aws:ecr:us-east-2::repository/repo-name" as unlike other resources above, ECR is region specific.

Reply:
it does not matter actually, because we don't have logics to check region and it is just a random string to check stdout again

Reply:
yeah it doesn't matter in our case (that's why it is bit). I just see you use complete ARNs for that other resources so pointed this out. It is actually missing both the region and accountId.